lgc/builder/ImageBuilder.cpp

```diff
@@ -1257,12 +1257,36 @@ Value *ImageBuilder::CreateImageAtomicCommon(unsigned atomicOp, unsigned dim, un
 Value *ImageBuilder::CreateImageQueryLevels(unsigned dim, unsigned flags, Value *imageDesc, const Twine &instName) {
   imageDesc = patchInvalidImageDescriptor(imageDesc);
   dim = dim == DimCubeArray ? DimCube : dim;
-  Value *zero = getInt32(0);
-  Instruction *resInfo = CreateIntrinsic(ImageGetResInfoIntrinsicTable[dim], {getFloatTy(), getInt32Ty()},
-                                         {getInt32(8), UndefValue::get(getInt32Ty()), imageDesc, zero, zero});
-  if (flags & ImageFlagNonUniformImage)
+
+  if (flags & ImageFlagNonUniformImage) {
```

Comment:
Why are you distinguishing here based on non-uniformity? Have you thought that through?